packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts

```diff
@@ -322,6 +346,15 @@ export class UserPoolClient extends Resource implements IUserPoolClient {
       }
     }
 
+    let tokenValidityUnits : CfnUserPoolClient.TokenValidityUnitsProperty | undefined;
+    if (props.accessTokenValidity || props.idTokenValidity || props.refreshTokenValidity) {
+      tokenValidityUnits = {
+        idToken: props.idTokenValidity ? 'minutes' : undefined,
+        accessToken: props.accessTokenValidity ? 'minutes' : undefined,
+        refreshToken: props.refreshTokenValidity ? 'minutes' : undefined,
+      };
+    }
+
```

Comment:
Could you also add validations around the allowed limits, and corresponding unit tests?

Might be cleaner to move all of it and computing `tokenValidityUnits` into a private method.